libraries/AP_Scripting/examples/CAN_logger.lua

```diff
@@ -0,0 +1,45 @@
+--[[
+   This script captures raw packets to a log file for later playback
+   Set CAN_D1_PROTOCOL to 10 for scripting.
+   Also need LOG_DISARMED set to 1 if running this while disarmed.
+--]]
+
+local can_driver = CAN:get_device(5)
+
+if not can_driver then
+   gcs:send_text(0,"No scripting CAN interface found")
+   return
+end
+
+local last_print_ms = millis()
+local frame_count = 0
+local last_frame_count = 0
+
+function update()
+
+   while true do
```

Comment:
You might find this times out if there is a fast enough data rate. You can do `return update 0` which will prevent timeouts and allow other scripts to run.

Reply:
I think the idea is that you'd mostly only run this script for a bench test, or for short test flights, so hopefully not with other scripts, and in any case, once it's detected a multi-frame packet (which, I think is what the while loop is for), you wouldn't want to let other scripts run in the middle of it, since it'll miss frames then, right? They'd get a chance to run once it hits that `break`, right?

Reply:
Agreed. You don't want something potentially blocking forever like this.

Reply:
> What do you mean by timeout?

The parameter `SCR_VM_I_COUNT` gives a max number of instructions before the script will be stopped, a bit like the watchdog for the main firmware (although your script will not be restarted).

You can give the script a better chance of not missing frames by increasing the buffer size it used, the 5 in `CAN:get_device(5)` is a frame buffer length. A easy fix might be to use a for loop rather than a while, just loop over the buffer length, that then gives a upper bound on the execution time.

Although as you say its a script meant for development, so I don't really think it matters so long as the limitations are known.

Reply:
i've increased the buffer to 25